### PR TITLE
split & tidy up the Invokers proposals, scoping for v1 invokers

### PR DESCRIPTION
**Pull request by keithamus**
This split the Invokers proposal into 2 separate sections; "Invokers" (what might also be known as `invoketarget`/`invokeaction` and related behaviour) and "Interest Invokers" (what might also be known as `interesttarget`). It also starts to demarcate chunks of the "Invokers" proposal into what we're initially shipping, vs what will come as a follow-up.

(I'm aware this is a big diff but... it's quite a bit to tease these apart, I can try and split it into smaller PRs if people would feel better about that?)

Closes https://github.com/openui/open-ui/issues/964, https://github.com/openui/open-ui/issues/956

### CODE REVIEW

File: site/src/pages/components/invokers.explainer.mdx
Comment (lukewarlow) [inline site/src/pages/components/invokers.explainer.mdx:118]:
```suggestion
dispatch InvokeEvents, allowing custom code to take control of
```

File: site/src/pages/components/invokers.explainer.mdx
Comment (lukewarlow) [inline site/src/pages/components/invokers.explainer.mdx:114]: Odd casing on invokeAction

**Comment by mfreed7** [comment]
I'm happy to review this (unfortunately not in the next few days, I'm otherwise swamped), but I wanted to ask: does this indicate anything about the urgency or priority of landing `interesttarget`? Personally, that's almost a bit more interesting (ha!) to me than `invoketarget`, because it unlocks tooltips and unblocks `popover=hint`. Are you still invested in the interest target proposal?

**Comment by keithamus** [comment]
There's no change in the prose that either prioritises or defers `interesttarget`. I'm still very much interested in pursuing it, it (hopefully) answers some as-yet-unanswered a11y patterns (see https://github.com/w3c/aria-practices/issues/128, I'm participating in the aria-wg in the hopes that we can thread this needle).

I'm hoping that we can ship `invoketarget` to use as learnings for `interesttarget`, scoping invokers down to dialog/popover pushes that forward so my hopes for `interesttarget` making progress are higher than they were last week 😉

**Comment by lukewarlow** [comment]
I very much hope we can get interest as a fast follow up. I do think it has more as yet unanswered questions but I'm hopeful we can make progress on it.

I believe in Chrome there's already a vast amount of code that can be reused, from the popover hint and hover experimentation.

**Comment by mfreed7** [comment]
Great, thank you both! I'm happy to assist with code reviews or whatever you need.

> I believe in Chrome there's already a vast amount of code that can be reused, from the popover hint and hover experimentation.

I would agree - Chromium has a full prototype of the hover-trigger and hover-detrigger code, plus CSS properties for controlling the delays, plus testing. That'll all likely need modification, and it doesn't handle non-mouse triggering, but it should hopefully be a good start.

**Comment by keithamus** [comment]
@mfreed7 could I get your eyes on this again? I think it'd be useful to separate these explainers so we can iterate more on the `interest` parts.

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (lukewarlow) [inline site/src/pages/components/interest-invokers.explainer.mdx:40]: Not sure if it's overly relevant but eye tracking causing interest events would be a potentially big privacy risk. This is why vision pro afaik explicitly doesn't trigger hover based on eye tracking. Interest would be the same.

Potentially worth calling out the risks or just removing references to eye tracking full stop?

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (lukewarlow) [inline site/src/pages/components/interest-invokers.explainer.mdx:58]:
```suggestion
as an available attribute to `<button>`, `<a>`,
```

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (lukewarlow) [inline site/src/pages/components/interest-invokers.explainer.mdx:62]: Idl needs updating

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (lukewarlow) [inline site/src/pages/components/interest-invokers.explainer.mdx:84]: Did this property get renamed on the invoke event? If so rename accordingly here

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (lukewarlow) [inline site/src/pages/components/interest-invokers.explainer.mdx:85]:
```suggestion
`InterestEvent`s are always non-bubbling, cancellable events.
```

Not sure what the correct way to do plurals of definitions is

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (lukewarlow) [inline site/src/pages/components/interest-invokers.explainer.mdx:91]: Is this the correct name? Above it said related target?

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (lukewarlow) [inline site/src/pages/components/interest-invokers.explainer.mdx:30]: Maybe call out the long press being overloaded and more thought needed on mobile for example chrome android shows a context menu

**Comment by lukewarlow** [comment]
I've left some comments but overall this looks good.

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (mfreed7) [reply site/src/pages/components/interest-invokers.explainer.mdx:30]: Agreed. I've been thinking that the exact methods for showing interest **need** to be UA-determined anyway. We can't spec eye-tracking. So for mobile, perhaps long-press followed by selection of something in UA-provided context menu?

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (mfreed7) [reply site/src/pages/components/interest-invokers.explainer.mdx:40]: ...or adding something like "eye tracking (potentially combined with another form of explicit confirmation such as a finger-tap to avoid privacy issues)..."? This, again, is where it's up to the UA to decide what constitutes "interest". The common examples of hovering with a mouse or keyboard-focusing should be very clear cases of interest, but others need UA help.

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (mfreed7) [inline site/src/pages/components/interest-invokers.explainer.mdx:73]: These are just the "attr-associated element steps" - no need for "popovertarget".

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (mfreed7) [inline site/src/pages/components/interest-invokers.explainer.mdx:78]: I don't love "Interestee" but I don't see a better alternative.

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (mfreed7) [inline site/src/pages/components/interest-invokers.explainer.mdx:142]: I'm guessing the a11y of this example is "bad". It's ok to include, but I bet you'll get comments. I wonder if there's something else you can do other than making content visible and invisible? Playing a sound or something?

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (mfreed7) [inline site/src/pages/components/interest-invokers.explainer.mdx:148]: I don't love the idea that the UA will determine the delays. That's probably complaint <span>#</span>1 about `<div title=foo>` - the delays are too long. My original proposal for popoverhovertarget had CSS-based delays. Any way to incorporate that?

File: site/src/pages/components/popover-hint.research.explainer.mdx
Comment (mfreed7) [inline site/src/pages/components/popover-hint.research.explainer.mdx:103]: nit: s/covers/cover

Comment (mfreed7) [reply site/src/pages/components/popover-hint.research.explainer.mdx:103]: And s/It/They

Comment (mfreed7) [reply site/src/pages/components/popover-hint.research.explainer.mdx:103]: And s/handles/handle.   Etc, I'll stop typing now.

**Review by mfreed7 — APPROVED** [review]
Sorry for the delay. Lots of comments, but this LGTM to land. We can iterate after landing.

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (keithamus) [reply site/src/pages/components/interest-invokers.explainer.mdx]: I've just removed this bit to unblock this PR. Perhaps we can discuss what shape this should take separately?

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (keithamus) [reply site/src/pages/components/interest-invokers.explainer.mdx:89]: Same :joy:

File: site/src/pages/components/interest-invokers.explainer.mdx
Comment (keithamus) [reply site/src/pages/components/interest-invokers.explainer.mdx]: Went with "`InterestEvent` objects"

**Comment by keithamus** [comment]
Thanks for the review from both! Aside from the various typos/nits, the changes of substance are:

 - Pulling out the descriptions on alternative HIDs and explicitly calling out that this is an in-progress area that's out of scope for the initial release.
 - Removing mention of adding delays, punting to another discussion/issue/PR.

**Comment by mfreed7** [comment]
> * Removing mention of adding delays, punting to another discussion/issue/PR.

That's ok by me, but let's definitely not forget about that part. I do think it's important.

**Pull request merged**
